coderd/database/queries/provisionerdaemons.sql

```diff
@@ -72,16 +75,30 @@ LEFT JOIN
 			LIMIT 1
 		)
 	)
+-- Current job information.
 LEFT JOIN
-	template_versions version ON version.id = (current_job.input->>'template_version_id')::uuid
+	workspace_builds current_build ON current_build.id = CASE WHEN current_job.input ? 'workspace_build_id' THEN (current_job.input->>'workspace_build_id')::uuid END
 LEFT JOIN
-	templates tmpl ON tmpl.id = version.template_id
+	-- We should always have a template version, either explicitly or implicitly via workspace build.
+	template_versions current_version ON current_version.id = CASE WHEN current_job.input ? 'template_version_id' THEN (current_job.input->>'template_version_id')::uuid ELSE current_build.template_version_id END
+LEFT JOIN
+	templates current_template ON current_template.id = current_version.template_id
+-- Previous job information.
+LEFT JOIN
+	workspace_builds previous_build ON previous_build.id = CASE WHEN previous_job.input ? 'workspace_build_id' THEN (previous_job.input->>'workspace_build_id')::uuid END
+LEFT JOIN
+	-- We should always have a template version, either explicitly or implicitly via workspace build.
+	template_versions previous_version ON previous_version.id = CASE WHEN previous_job.input ? 'template_version_id' THEN (previous_job.input->>'template_version_id')::uuid ELSE previous_build.template_version_id END
+LEFT JOIN
+	templates previous_template ON previous_template.id = previous_version.template_id
```

Comment:
Review: I completed this because it was missing some things. But I wonder if we need to take RBAC into consideration now?

The endpoint previously returned a template version ID or workspace build ID only, not it also returns information regarding the template that job is associated with.

This has some implications considering regular members are allowed to view provisioner daemon resources. WDYT @johnstcn?

I'm not particularly fond of it, but we could add the same authorization to the API endpoint as for jobs:

```go
	// For now, only owners and template admins can access provisioner jobs.
	if !api.Authorize(r, policy.ActionRead, rbac.ResourceProvisionerJobs.InOrg(org.ID)) {
		httpapi.ResourceNotFound(rw)
		return nil, false
	}
```

Reply:
I don't know how sensitive `template` or `template_version` fields can be, but a red flag for me would be if this could cross org boundaries.

I think some further questions we need to answer before we can even update the RBAC rules are:
- Who can view a template import job not linked to any workspace? (Suggestion: anyone who can read the linked template)
- Who can view a workspace build job? (Suggestion: anyone who can read the linked workspace)

To be safe, I think we should promote the same level of access control as for provisioner jobs _for now_. We should add a follow-up issue to take provisioner job ownership and template-level ACLs into account.

Does that sound reasonable to you?